cypress/e2e/yves/core/locale-switching.cy.ts

```diff
@@ -0,0 +1,57 @@
+import { container } from '@utils';
+import { LocaleSwitchingScenario } from '@scenarios/yves';
+import { CatalogPage, HomePage } from '@pages/yves';
+import { LocaleStaticFixtures } from '@interfaces/yves';
+
+(['b2c', 'b2c-mp', 'b2b', 'b2b-mp'].includes(Cypress.env('repositoryId')) ? describe.skip : describe)(
+  'locale switching',
+  { tags: ['@core', '@yves'] },
+  (): void => {
+    const homePage = container.get(HomePage);
+    const catalogPage = container.get(CatalogPage);
+    const localeSwitchingScenario = container.get(LocaleSwitchingScenario);
+
+    let staticFixtures: LocaleStaticFixtures;
+
+    before((): void => {
+      ({ staticFixtures } = Cypress.env());
+    });
+
+    /**
+     * Helper method for testing locale switching on any page.
+     * @param visitPage - Function to visit the page (e.g., `catalogPage.visit`).
+     * @param page - The page object containing methods like `getAvailableLocales`.
+     */
+    const testLocaleSwitching = (visitPage: () => void, page: { getAvailableLocales: () => void }): void => {
+      visitPage();
+
+      page.getAvailableLocales();
+
+      localeSwitchingScenario.execute({
+        currentLocale: staticFixtures.localeEN,
+        selectedLocale: staticFixtures.localeDE,
+      });
+
+      cy.reload();
+      localeSwitchingScenario.execute({
+        currentLocale: staticFixtures.localeDE,
+        selectedLocale: staticFixtures.localeEN,
+      });
+    };
+
+    it('Should be able to switch locales at the home page.', (): void => {
+      testLocaleSwitching(() => homePage.visit(), homePage);
+    });
+
+    it('Should be able to switch locales at the catalog page.', (): void => {
+      testLocaleSwitching(() => catalogPage.visit(), catalogPage);
+    });
+
+    it('Should be able to switch locales at the product detailed page.', (): void => {
+      testLocaleSwitching(() => {
+        catalogPage.visit();
+        catalogPage.search({ query: staticFixtures.productName });
```

Comment:
1. Please check if we really need to have here `catalogPage.visit();` as the next step is 'perform search and view search results page'. I want to avoid useless actions (load search/catalog page 2 times). I know that in other tests those 2 comes together, but it might be a mistake.
2. `catalogPage.search` doesn't open PDP. The result of this method is search result page (catalog), not PDP. Your test is about switching locale on PDP.
3. Try to avoid staticFixtures when it comes to products. Static fixtures might be used only in the test case that has a hard dependency on demodata -> looks like this is not our case, so please use dynamic fixture for products.

Reply:
1. Yes we need it, I've checked and `.search` won't work without that step.
2. It opens. It searches the product by name and then goes to the first item in the catalog by its URL. 
3. Thanks, will fix it